### PR TITLE
Provide jpa metamodel for PanacheEntity class

### PR DESCRIPTION
**Pull request by Postremus**
The created meta model for an entity extends PanacheEntity_.
PanacheEntity_ is not created during the quarkus build, and therefore would have to be created by the user, which is tedius.

We now just run the jpamodelgen in the hiberate-orm-panache modul, creating PanacheEntity_ automaticly during the build.

### CODE REVIEW

File: extensions/panache/hibernate-orm-panache/runtime/pom.xml
Comment (gsmet) [inline extensions/panache/hibernate-orm-panache/runtime/pom.xml]:
```suggestion
            <!-- Exclude these dependencies, since they are banned -->
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks.

I'll let @FroMage decide but I think it makes sense to do it. Could you apply my suggestion and squash everything?

File: extensions/panache/hibernate-orm-panache/runtime/pom.xml
Comment (Postremus) [reply extensions/panache/hibernate-orm-panache/runtime/pom.xml]: sure, I will do this change this evening.

**Comment by FroMage** [comment]
I've no idea what this will do. What's the JPA meta-model, and why do we need to generate it?

**Comment by Postremus** [comment]
@FroMage See Example 1 and Example 2 from https://docs.jboss.org/hibernate/orm/5.0/topical/html/metamodelgen/MetamodelGenerator.html.

The metamodel allows you to simplify your criteriaquery.

For example, I can write:
`where.add(builder.not(historyEntry.get(HistoryEntry_.eventType).in(filter.getExlusiveEventTypes())));`

Instead of:
`where.add(builder.not(historyEntry.get("eventType").in(filter.getExlusiveEventTypes())));`.

This saves me from referencing "entityType" as String everywhere.
It also allows type safety.

**Comment by FroMage** [comment]
OK, thanks. I guess it can't hurt, but I'm not familiar with this at all, so asking @Sanne and @emmanuelbernard for advice.

**Comment by gsmet** [comment]
Yeah, it's useful if you use JPA criterias. If you have nothing against it, I will merge it once the typo is fixed.

**Comment by Sanne** [comment]
Nice!

**Comment by Postremus** [comment]
@gsmet done

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Comment by geoand** [comment]
I think it would be nice to have a section in the Panache guide showing how this could be used? I personally don't know much about the meta-model and would like to know how to use it :)

**Comment by emmanuelbernard** [comment]
Nan, maybe in the normal Hibernate doc. This tool should be in the
Hibernate ORM extension anyways, not in the Panache one as it's generally
useful.

**Comment by gsmet** [comment]
The idea here is to provide the meta model for `PanacheEntity`.

**Comment by geoand** [comment]
> The idea here is to provide the meta model for `PanacheEntity`.

Exactly. So perhaps some quick info on how to use it with Panache and then perhaps a link to the extensive Hibernate documentation?

**Pull request merged**
